### DISCUSSION TITLE
Support for filter and search on collection views in url for easy share

### DISCUSSION THREAD

**Discussion by u12206050** [directus/directus#16228] [Feature Requests]
In order to easily share collection results after setting up filters and a search query, I propose syncing the filter and search value into the url allowing the user to copy and send that to another user to simply open and have the filter and search value populated from the url.

It is not the idea to replace presets, but rather to make it easy to share the setup of filters with others. Example; someone less technical could ask a developer or someone that knows for a certain subset of data. The person asked, could search and filter the collection on their side, copy the link send it across and have the other user view it (within their rights obviously)

Since permissions might differ, there isn't a need for sharing `columns`, `layouts` etc. if that is a need, a preset could rather be created and shared (somehow)

Possible PR already implemented here: #14081 

![](https://user-images.githubusercontent.com/6641242/190096427-872a2e0f-18e8-4546-9d7e-d3da291538ca.gif)

**Comment by u12206050** [comment]
One benefit of updating the filter and search in query is also saving it as part of the history state, so that when you go to a bookmark and slightly change it, you can view items and go back to your slightly changed bookmark, without it resetting to the original values, causing you to recreate the filter.

**Comment by Saissaken** [comment]
Excellent work @u12206050 !

I been looking at your PR, and it's exactly what I need to cover a use case for a client, were creating bookmarks is not an accepted solution, as won't scale.
Sadly your PR it's not merged yet, so I guess I'll need to use a fork of directus on prod 👀, hoping that this feature gets merged soon! As there is no workaround AFAIK.

I found myself many times in the need of providing a dynamic way to filter a table, without creating a bookmark, and without the user having to filter the view manually.
I agree that URL parameters are the way to go for this type of use cases.

Idea:
For those cases where we need to apply filters while maintaining a layout, maybe it would be possible to inherit the layout from an existing bookmark, and then apply filters on top of that.
Example:

```
/admin/content/sample_table?bookmark=25&filter={....}
```

Thanks for your work!

**Reply by Saissaken** [reply]
Note: I was viewing again the gif you attached and just noticed that the behavior I've explain it's already implemented on your PR, well done!

**Comment by Saissaken** [comment]
Here is another discussion with similar requirements, it's worth the read #6493

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
